tests/unit/backend/service/test_CalibrationService.py

```diff
@@ -1151,6 +1152,33 @@ def test_hasState(self):
         )
         assert not self.instance.hasState(badRequest)
 
+    @mock.patch("snapred.backend.service.CalibrationService.CalculateResidualDiffCalRecipe")
+    def test_calculateResidual(self, MockCalculateResidualDiffCalRecipe):
+        # Arrange: set up mock request and mock recipe
+        mockRequest = mock.Mock(
+            spec=CalculateResidualRequest,
+            inputWorkspace="inputWS",
+            outputWorkspace="outputWS",
+            fitPeaksDiagnostic="fitPeaksDiagWS",
+        )
+
+        # Mock the recipe execution result
+        mockRecipeInstance = MockCalculateResidualDiffCalRecipe.return_value
+        mockRecipeInstance.executeRecipe.return_value = "expected_result"
+
+        # Act: call calculateResidual with the mock request
+        result = self.instance.calculateResidual(mockRequest)
+
+        # Assert: check that the recipe was called with correct parameters
+        MockCalculateResidualDiffCalRecipe.assert_called_once_with()
+        mockRecipeInstance.executeRecipe.assert_called_once_with(
+            InputWorkspace=mockRequest.inputWorkspace,
+            OutputWorkspace=mockRequest.outputWorkspace,
+            FitPeaksDiagnosticWorkSpace=mockRequest.fitPeaksDiagnostic,
+        )
+        # Assert that the result is what the recipe returns
+        self.assertEqual(result, "expected_result")  # noqa: PT009
```

Comment:
I have been trying to transfer tests to use [`mock.sentinel`](https://www.seanh.cc/2017/03/17/sentinel/).  I think it should be used here instead of `"inputWS"`, `"outputWS"`, `"fitPeaksDiagWS"`, and `"expected_result"`.

Reply:
Tried to implement this, but couldn't get it to work properly.